### DISCUSSION TITLE
ECMAScript란?

### DISCUSSION THREAD

**Discussion by al-bur** [woowacourse-study/2022-woowahan-bansanghwe#136] [JavaScript]
## ECMAScript란 무엇인가?

ECMA스크립트란, Ecma International이 **ECMA-262** 기술 규격에 따라 정의하고 있는 표준화된 스크립트 프로그래밍 언어를 말한다. 자바스크립트를 표준화하기 위해 만들어졌다.

ECMAScript도 Javascript처럼 언어이다. 

어떤 블로그에서는 아래와 같이 묘사를 해주고 있다. Javascript, JScript와 같은 언어들이 자동차라면, ECMAScript라는 동일한 엔진을 사용한다라고 표현하고있다.

> Languages such as ActionScript, JavaScript, JScript all use ECMAScript as its core. As a comparison, AS/JS/JScript are 3 different cars, but they all use the same engine…

<br>

## 무엇을 해결하기 위해 나왔을까?

결론을 말하면, 제각기 브라우저에서 사용하는 스크립트 언어가 달랐기 때문에 이를 표준화하기 위해서 나왔다. 

과거에는, Internet Explorer와 Netscape Navigator 브라우저에서 사용하는 스크립팅 언어가 JScript와 Javascript로 나뉘어져있었다. 두 개의 브라우저는 각각 다른 언어를 사용했지만, 통일할 계획이 없었고, 심지어 IE에서 잘보이는 웹사이트, Navigator에서 잘보이는 웹사이트라는 표시가 생기게 되었다.

이러한 경우들이 발생해서, Netscape Navigator가 ECMA에 javascript를 가지고 표준을 만들어달라고 부탁을 하게 된다.

> In November of 1996, Netscape submitted JavaScript to Ecma to build out a standard specification.
> 

**그렇다면, 왜 javascript가 아닌 ECMAScript란 이름을 사용한 것일까?**

이는, javascript라는 이름에 대한 상표를 오라클이 가지고 있었기 때문에, 법적인 문제를 피하기 위해 ECMAScript로 이름을 변경했다고 한다.

**Comment by greenblues1190** [comment]
ECMAScript에 스펙이 추가되는 과정?

**Comment by greenblues1190** [comment]
프론트엔드에서 백엔드와 다르게 최신 스펙을 적극적으로 사용하는 이유?

**Comment by greenblues1190** [comment]
ECMAScript 역사에서 기억에 남는 변경점들

**Comment by greenblues1190** [comment]
최신 ECMAScript 뉴스는 어디서 얻으세요?

**Comment by JUDONGHYEOK** [comment]
타입스크립트를 효과적으로 사용하기 위한 본인만의 철학?

**Comment by JUDONGHYEOK** [comment]
자바스크립트는 왜 var만 사용했을까?

**Comment by JUDONGHYEOK** [comment]
타입스크립트를 다른 정적 타입 언어와 비교하면?
